autofs-5.1.1 - fix direct mount stale instance flag reset

From: Ian Kent <raven@themaw.net>

When re-reading a direct map the stale map instance flag was not being
reset. This caused the map to be seen as stale on subsequent lookups
thereby triggering a map re-read.

Signed-off-by: Ian Kent <raven@themaw.net>
---
 CHANGELOG      |    1 +
 daemon/state.c |    1 +
 2 files changed, 2 insertions(+)

--- autofs-5.0.7.orig/CHANGELOG
+++ autofs-5.0.7/CHANGELOG
@@ -178,6 +178,7 @@
 - make find_dc_server() return a status.
 - make find_server() return a status.
 - fix return handling of do_reconnect() in ldap module.
+- fix direct mount stale instance flag reset.
 
 25/07/2012 autofs-5.0.7
 =======================
--- autofs-5.0.7.orig/daemon/state.c
+++ autofs-5.0.7/daemon/state.c
@@ -513,6 +513,7 @@ static void *do_readmap(void *arg)
 			}
 			lookup_prune_one_cache(ap, map->mc, now);
 			pthread_cleanup_pop(1);
+			clear_stale_instances(map);
 			map->stale = 0;
 			map = map->next;
 		}
